docs/modules/clients/pages/java.adoc

```diff
@@ -1715,205 +1447,140 @@ YAML::
-connectionRetryConfig.setInitialBackoffMillis(1000)
-                     .setMaxBackoffMillis(60000)
-                     .setMultiplier(2)
-                     .setClusterConnectTimeoutMillis(50000)
-                     .setJitter(0.2);
-
+ClientConfig clientConfig = new ClientConfig();
+clientConfig.getNetworkConfig().setConnectionTimeout(5000);
 ----
 
-The following are configuration element descriptions:
+The default value is *5000* milliseconds.
 
-* `initial-backoff-millis`: Specifies how long to wait (backoff), in milliseconds, after the first failure before retrying.
-Its default value is 1000 ms.
-* `max-backoff-millis`: Specifies the upper limit for the backoff in milliseconds.
-Its default value is 30000 ms.
-* `multiplier`: Factor to multiply the backoff after a failed retry.
-Its default value is 1.05.
-* `cluster-connect-timeout-millis`: Timeout value in milliseconds for the client to give up
-to connect to the current cluster. Its default value is `-1`, i.e., infinite.
-For the default value, client will not stop trying to
-connect to the target cluster (infinite timeout). If the failover client is used
-with the default value of this configuration element, the failover client will try
-to connect alternative clusters after 120000 ms (2 minutes). For any other value,
-both the client and the failover client will use this as it is.
-* `jitter`: Specifies by how much to randomize backoff periods. Its default value is 0.
+=== Set a socket interceptor
 
-A pseudo-code is as follows:
+[blue]*Hazelcast {enterprise-product-name}*
 
-[source,shell]
+Any class implementing `com.hazelcast.nio.SocketInterceptor` is a socket interceptor.
+The following code sample shows an example of how to set a socket interceptor:
+
+[source,java]
 ----
- begin_time = getCurrentTime()
- current_backoff_millis = INITIAL_BACKOFF_MILLIS
- while (TryConnect(connectionTimeout)) != SUCCESS) {
-    if (getCurrentTime() - begin_time >= CLUSTER_CONNECT_TIMEOUT_MILLIS) {
-         // Give up to connecting to the current cluster and switch to another if exists.
-         // For the default values, CLUSTER_CONNECT_TIMEOUT_MILLIS is infinite for the
-         // client and equal to the 120000 ms (2 minutes) for the failover client.
-    }
-    Sleep(current_backoff_millis + UniformRandom(-JITTER * current_backoff_millis, JITTER * current_backoff_millis))
-    current_backoff = Min(current_backoff_millis * MULTIPLIER, MAX_BACKOFF_MILLIS)
+public interface SocketInterceptor {
+    void init(Properties properties);
+    void onConnect(Socket connectedSocket) throws IOException;
 }
 ----
 
-Note that, `TryConnect` above tries to connect to any member that the client knows,
-and for each connection we have a connection timeout; see the
-<<setting-connection-timeout, Setting Connection Timeout section>>.
+The first step initializes the `SocketInterceptor` using the defined properties.
+The second step informs when the socket is connected using the `onConnect` method.
```

Comment:
Why do we use the word "step" here? Are we talking about some internal Hazelcast member process steps?  Or are we trying to talk about the interface methods? If it's about methods, we should call them methods, and  describe as 

"the first method is called when the SocketInterceptor is being initailized" 
"the second method is called when a socket is connected"

Reply:
I agree this could be improved - its referring to void init and void onConnect right?

```suggestion
The first method initializes the `SocketInterceptor` using the defined properties.
The second method informs when the socket is connected using the `onConnect` method.
```

Reply:
yes I think so, the change looks fine to me .